hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/HoodieOptionConfig.scala

```diff
@@ -67,21 +68,18 @@ object HoodieOptionConfig {
     .withSqlKey("payloadClass")
     .withHoodieKey(DataSourceWriteOptions.PAYLOAD_CLASS_NAME.key)
     .withTableConfigKey(HoodieTableConfig.PAYLOAD_CLASS_NAME.key)
-    .defaultValue(DataSourceWriteOptions.PAYLOAD_CLASS_NAME.defaultValue())
     .build()
 
-  val SQL_PAYLOAD_TYPE: HoodieSQLOption[String] = buildConf()
-    .withSqlKey("payloadType")
-    .withHoodieKey(DataSourceWriteOptions.PAYLOAD_TYPE.key)
-    .withTableConfigKey(HoodieTableConfig.PAYLOAD_TYPE.key)
-    .defaultValue(DataSourceWriteOptions.PAYLOAD_TYPE.defaultValue())
+  val SQL_MERGE_MODE: HoodieSQLOption[String] = buildConf()
+    .withSqlKey("mergeMode")
+    .withHoodieKey(HoodieWriteConfig.RECORD_MERGE_MODE.key)
+    .withTableConfigKey(HoodieTableConfig.RECORD_MERGE_MODE.key)
     .build()
 
   val SQL_RECORD_MERGER_STRATEGY: HoodieSQLOption[String] = buildConf()
     .withSqlKey("recordMergerStrategy")
-    .withHoodieKey(DataSourceWriteOptions.RECORD_MERGER_STRATEGY.key)
+    .withHoodieKey(DataSourceWriteOptions.RECORD_MERGER_STRATEGY_ID.key)
```

Comment:
Change `HoodieTableConfig.RECORD_MERGER_STRATEGY` to `HoodieTableConfig.RECORD_MERGER_STRATEGY_ID` without changing the config key?